cmake/Platform/Core/BoardFlags/CompilerFlagsSetter.cmake

```diff
@@ -1,22 +1,30 @@
 # ToDo: Comment
 function(set_board_compiler_flags COMPILER_FLAGS ARDUINO_DEFINED_VERSION BOARD_ID IS_MANUAL)
 
-    set(COMPILE_FLAGS "-DF_CPU=${${BOARD_ID}.build.f_cpu} -DARDUINO=${ARDUINO_DEFINED_VERSION} -mmcu=${${BOARD_ID}.build.mcu}")
-    if (DEFINED ${BOARD_ID}.build.vid)
-        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_VID=${${BOARD_ID}.build.vid}")
+    _get_board_property(${BOARD_ID} build.f_cpu FCPU)
+    _get_board_property(${BOARD_ID} build.mcu MCU)
+    set(COMPILE_FLAGS "-DF_CPU=${FCPU} -DARDUINO=${ARDUINO_DEFINED_VERSION} -mmcu=${MCU}")
+
+    _get_board_property_if_exists(${BOARD_ID} build.vid VID)
+    _get_board_property_if_exists(${BOARD_ID} build.pid PID)
+    if (VID)
+        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_VID=${VID}")
     endif ()
-    if (DEFINED ${BOARD_ID}.build.pid)
-        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_PID=${${BOARD_ID}.build.pid}")
+    if (PID)
+        set(COMPILE_FLAGS "${COMPILE_FLAGS} -DUSB_PID=${PID}")
     endif ()
+
     if (NOT IS_MANUAL)
+        _get_board_property(${BOARD_ID} build.core BOARD_CORE)
         set(COMPILE_FLAGS "${COMPILE_FLAGS} -I\"${${BOARD_CORE}.path}\" -I\"${ARDUINO_LIBRARIES_PATH}\"")
         if (${ARDUINO_PLATFORM_LIBRARIES_PATH})
             set(COMPILE_FLAGS "${COMPILE_FLAGS} -I\"${ARDUINO_PLATFORM_LIBRARIES_PATH}\"")
         endif ()
     endif ()
     if (ARDUINO_SDK_VERSION VERSION_GREATER 1.0 OR ARDUINO_SDK_VERSION VERSION_EQUAL 1.0)
         if (NOT IS_MANUAL)
-            set(PIN_HEADER ${${${BOARD_ID}.build.variant}.path})
+            _get_board_property(${BOARD_ID} build.variant VARIANT)
+            set(PIN_HEADER ${${VARIANT}.path})
```

Comment:
It seems like a little break of consistency, isn't it?
You've read many properties of a board, yet left out the **path** property. I honestly don't know if this is intentional, or if it even matters, just would like to get your opinion on this one.

Reply:
It is correct, it reads pinlayout type from board property (e.g. standard) and constructs include path to arduino sdk: e.g. /hardware/arduino/avr/variants/standard

Reply:
Yes, I agree with that this behavior is correct.
What I don;t understand is why you left the `path` property to be explicitly stated.
To further explain myself, let's assume that you *have** implemented what I'm talking about. It would look like this: `_get_property_board(${BOARD_ID} build.variant.path)`

Reply:
I know that you mean, but there is a catch: *board.variant.path* doesn't exist i board properties.
In *register_hardware_platform* function cmake caches all paths to pinheaders. (If you would look at cmake cache you should see keys like: standard.path, eightanalogpins.path, mega.path, etc) so i need load name of variant e.g. standard, and then read cached cmake value standard.path.

Reply:
I still can't understand how is it different from any other property you've supported using the `_get_board_property` function. If the `variant.path` is stored in cache, it's exactly the same as if simply `variant` is stored in cache, isn't it?
I'm sorry if I get you wrong, and as a matter of fact it's really not that important, but closing this now before we merge those changes seems correct for me.

Reply:
just play with it :) 
look at */hardware/arduino/avr* folder, at boards.txt and in cmake cached values, and you'll get it :)

Reply:
I finally got it. Thanks and sorry for disturbing you so much with it :)